pywbem/_subscription_manager.py

```diff
@@ -1284,16 +1301,60 @@ def _create_filter(self, server_id, source_namespace, query,
                         filter_inst = server.conn.GetInstance(filter_path)
                         self._owned_filters[server_id][i] = filter_inst
                     return filter_inst
-            filter_path = server.conn.CreateInstance(filter_inst)
+
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
             self._owned_filters[server_id].append(filter_inst)
         else:
             # Responsibility to ensure it does not exist yet is with the user
-            filter_path = server.conn.CreateInstance(filter_inst)
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
 
+        # If systemname does not yet exist, create it
+        if server_id not in self._systemnames:
+            self._systemnames[server_id] = filter_inst['SystemName']
+        else:
+            assert self._systemnames[server_id] == filter_inst['SystemName']
```

Comment:
This assert depends on server behavior. What if the server is inconsistent across different providers? Maybe issuing a Python warning would be more robust.